### PR TITLE
[INTEL MKL] Add 2D Concat Feature with MKL-DNN

### PR DESCRIPTION
**Pull request by mdfaijul**
This PR adds a feature for concatenating 2D tensors with MKL-DNN. Such concatenation is used in Wide and Deep models.

### CODE REVIEW

**Comment by mdfaijul** [comment]
Temporarily closing. Will reopen after some more changes.

**Comment by mdfaijul** [comment]
@penpornk did you have chance to look into this PR?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm so sorry for my delay!

File: tensorflow/core/kernels/mkl_concat_op.cc [hunk 1]
```diff
@@ -352,31 +356,36 @@ class MklConcatOp : public OpKernel {
...
+        if (dst_dims.size() == 4) {
+          dst_dims_in_nchw = MklDnnDimsInNCHW(
+              dst_dims, MklDnnDataFormatToTFDataFormat(orig_tf_format));
+          // Set the output format same as the most common format of inputs
+          // to avoid layout conversions.
+          dst_md = memory::desc(dst_dims_in_nchw, MklDnnType<T>(),
+                                mkl_common_format);
+        } else if (dst_dims.size() == 2 &&
+                   mkl_common_format == memory::format::nc) {
+          // When memory::format::nc, dst_dims are already in MKLDNN order
```
[the remark this answers is not in the document]
Comment (mdfaijul) [reply tensorflow/core/kernels/mkl_concat_op.cc]: @penpornk Addressed all your comments. Please check.

**Review by penpornk — APPROVED** [review]
Thank you again for contributing!

**Pull request merged**